game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/AaCasualtySelectorTest.java

```diff
@@ -0,0 +1,432 @@
+              mock(Territory.class));
+
+      assertThat("5 planes should be killed or damaged", details.size(), is(5));
+    }
+  }
+
+  @Nested
+  class IndividualRollCasualties {
+    private IDelegateBridge bridge;
+
+    @BeforeEach
+    void initializeGameData() {
+      gameData =
+          givenGameData()
+              .withDiceSides(6)
+              .withEditMode(false)
+              .withChooseAaCasualties(false)
+              .withLowLuck(false)
+              .withLowLuckAaOnly(false)
+              .withRollAaIndividually(true)
+              .build();
+      bridge = mock(IDelegateBridge.class);
+      when(bridge.getData()).thenReturn(gameData);
+
+      aaUnitType = new UnitType("aaUnitType", gameData);
+      final UnitAttachment aaUnitAttachment =
+          new UnitAttachment("aaUnitAttachment", aaUnitType, gameData);
+      aaUnitType.addAttachment(UNIT_ATTACHMENT_NAME, aaUnitAttachment);
+
+      damageableAaUnitType = new UnitType("aaUnitType", gameData);
+      final UnitAttachment damageableAaUnitAttachment =
+          new UnitAttachment("damageableAaUnitAttachment", aaUnitType, gameData);
+      damageableAaUnitAttachment.setDamageableAa(true);
+      damageableAaUnitType.addAttachment(UNIT_ATTACHMENT_NAME, damageableAaUnitAttachment);
+
+      planeUnitType = new UnitType("planeUnitType", gameData);
+      final UnitAttachment planeUnitAttachment =
+          new UnitAttachment("planeUnitAttachment", aaUnitType, gameData);
+      planeUnitType.addAttachment(UNIT_ATTACHMENT_NAME, planeUnitAttachment);
+
+      planeMultiHpUnitType = new UnitType("planeUnitType", gameData);
+      final UnitAttachment planeMultiHpUnitAttachment =
+          new UnitAttachment("planeMultiHpUnitAttachment", aaUnitType, gameData);
+      planeMultiHpUnitAttachment.setHitPoints(2);
+      planeMultiHpUnitType.addAttachment(UNIT_ATTACHMENT_NAME, planeMultiHpUnitAttachment);
+    }
+
+    @Test
+    void hitsEqualToPlanesKillsAll() {
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              givenPlaneUnits(1),
+              givenAaUnits(1),
+              mock(CombatValue.class),
+              givenAaCombatValue(),
+              "text",
+              givenDiceRoll(true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat("One plane was hit and killed", details.getKilled(), hasSize(1));
+      assertThat(details.getDamaged(), is(empty()));
+    }
+
+    @Test
+    void hitsLessThanPlanesKillsAccordingToTheRolledDice() {
+
+      final List<Unit> planes = givenPlaneUnits(5);
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              planes,
+              givenAaUnits(1),
+              mock(CombatValue.class),
+              givenAaCombatValue(),
+              "text",
+              givenDiceRoll(true, false, true, false, true),
```

Comment:
This 'given' method is not easy to understand. A javadoc on the given method would help avoid having to reverse engineer the code. The name of the method probably ought to describe what kind of dice roll is being returned. EG: `givenDiceRollsWithHitSequence`, or something like that.